update.sh

```diff
@@ -1,218 +1,269 @@
+    shift # past argument
+    ;;
+esac
+done
+
+set -- "${POSITIONAL[@]}" # restore positional parameters
+
+
+if [[ -z $QUIET_MODE ]];
+then
+
 echo ""
 echo "This will update the Pacefactory SCV2 deployment running on this machine."
 echo ""
-echo "You will be prompted to enable optional services."
-echo "For each prompt, you may enter 'y', 'n', or '?'"
-echo "corresponding to yes, no, and help, respectively."
-echo "If you are unsure of the importance of given service, select the '?' option."
-echo ""
-echo ""
-echo "The final prompt will ask if you are running in ONLINE or OFFLINE mode."
-echo ""
-echo "ONLINE mode is to be used for client sites where the autodelete feature"
-echo "must be enabled to manage storage constraints."
-echo "OFFLINE mode is to be used when running videos using the Offline Processing"
-echo "tool. This disabled the autodelete feature, so data persists in the dbserver."
 
-# Init profiles string as empty (no optional profiles)
-profile_str=""
-# Init override string with base docker-compose and override docker-compose file
-override_str="-f docker-compose.yml -f docker-compose.override.yml"
-DEBUG=true
+fi
 
-PROJECT_NAME=$1
+ENV_FILE=".env"
 
-echo ""
-if [[ -z $PROJECT_NAME ]];
+if [ ! -f .env ];
 then
-    DEFAULT_PROJECT=deployment-scripts
-    CURRENT_PROJECT=$(docker compose ls --all --quiet | head -1)
-    PROJECT_NAME=${CURRENT_PROJECT:-$DEFAULT_PROJECT}
+    echo ".env file not found. Using .env.example"
+    echo ""
+    ENV_FILE=".env.example"
+fi
 
-    read -r -p "Confirm project name [$PROJECT_NAME]: "
-    if [[ ! -z "$REPLY" ]];
-    then
-        PROJECT_NAME=$REPLY
-    fi;
+DEFAULT_PROJECT=deployment-scriptsl
+CURRENT_PROJECT=$(docker compose ls --all --quiet | head -1)
+CURRENT_PROJECT=${CURRENT_PROJECT:-$DEFAULT_PROJECT}
+PROJECT_NAME=${PROJECT_NAME:-CURRENT_PROJECT}
+
+if [[ -z $QUIET_MODE ]];
+then 
+  read -r -p "Confirm project name [$PROJECT_NAME]: "
+  if [[ ! -z "$REPLY" ]];
+  then
+      PROJECT_NAME=$REPLY
+  fi;
 fi
-echo "Project name: '$PROJECT_NAME'"
 
-# Enable social profile?
+echo "Project name: '$PROJECT_NAME'"
 echo ""
-while read -r -p "Enable the social profile? (y/[n]/?) "
+
+override_str="-f docker-compose.yml"
+profile_str=""
+
+if [[ -z $QUIET_MODE ]];
+then 
+  echo "You will be prompted to enable optional services."
+  echo "For each prompt, you may enter 'y', 'n', or '?'"
+  echo "corresponding to yes, no, and help, respectively."
+  echo "If you are unsure of the importance of given service, select the '?' option."
+fi
+
+for profile_compose_file in docker-compose.*.yml
 do
-    if [[ "$REPLY" == "y" ]];
-    then
-        profile_str="$profile_str --profile social"
-        echo " -> Will enable social profile"
-        break;
-    elif [[ "$REPLY" == "?" ]];
+    profile_id="${profile_compose_file#docker-compose.}"
+    profile_id=${profile_id%.yml}
+
+    name=$(cat $profile_compose_file | yq -e -r '.["x-pf-info"].name // empty')
```

Comment:
My version of `yq` doesn't seem to have a `-r` flag. What flag is that supposed to represent?

Reply:
Yeah, the packaged versions don't seem to be the same, you need the one from pypi: https://pypi.org/project/yq/

Reply:
Also the `-r` flag:

> `--raw-output` / `-r`:
> With this option, if the filter's result is a string then it will be written directly to standard output rather than being formatted as a JSON string with quotes